### ISSUE TITLE
 API Doc only access for team members in Supabase UI

### ISSUE THREAD

**Issue by thestepafter** [supabase/supabase#9029]
# Feature request

## Is your feature request related to a problem? Please describe.

My front-end developers only need access to the API Docs in the Supabase UI but I do not see a way to limit their access to just the API Docs.

I am basing this assumption off of the information in [https://supabase.com/docs/guides/hosting/platform](https://supabase.com/docs/guides/hosting/platform).

## Describe the solution you'd like

Ideally there would be a permission that you can grant to team members which only provides API Docs access. No SQL editor or other access.

## Describe alternatives you've considered

Cut and pasting all relevant API Docs to a different tool for my front-end developers to access or not using Supabase.

## Additional context

This is for security and data integrity reasons and not something I am able to compromise on by giving more access than the API Docs.

**Comment by saltcod** [comment]
Hey @thestepafter Do you mean you'd like to have your devs only be able to access: 
https://app.supabase.com/project/-/api

And not tools like the table editor / sql editor / storage, etc?

**Comment by thestepafter** [comment]
@saltcod yes that is correct, no access to table editor, sql editor, or storage, etc. Just an API documentation only view. (This would be very helpful for us.)

**Comment by kiwicopple** [comment]
Hey @thestepafter - we have a set of broad permissions here: 
https://supabase.com/docs/guides/hosting/platform#manage-team-members

We will bring out "role customization" eventually, but be warned that probably require an upgrade. For now, I'll note down your feedback with the team. It's a bit easier for us to manage a few roles for now, and we'll role out more granular permissions as we layer in audit logs etc
